pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -33,14 +105,458 @@ benchmarks! {
 		};
 
 	}: _(RawOrigin::Root, inflation_range)
+	verify {}
+
+	set_parachain_bond_account {
+		let parachain_bond_account: T::AccountId = account("TEST", 0u32, USER_SEED);
+	}: _(RawOrigin::Root, parachain_bond_account.clone())
+	verify {
+		assert_eq!(Pallet::<T>::parachain_bond_info().account, parachain_bond_account);
+	}
+
+	set_parachain_bond_reserve_percent {
+	}: _(RawOrigin::Root, Percent::from_percent(33))
+	verify {
+		assert_eq!(Pallet::<T>::parachain_bond_info().percent, Percent::from_percent(33));
+	}
+
+	set_total_selected {}: _(RawOrigin::Root, 100u32)
+	verify {
+		assert_eq!(Pallet::<T>::total_selected(), 100u32);
+	}
+
+	set_collator_commission {}: _(RawOrigin::Root, Perbill::from_percent(33))
+	verify {
+		assert_eq!(Pallet::<T>::collator_commission(), Perbill::from_percent(33));
+	}
+
+	set_blocks_per_round {}: _(RawOrigin::Root, 1200u32)
+	verify {
+		assert_eq!(Pallet::<T>::round().length, 1200u32);
+	}
+
+	// USER DISPATCHABLES
+
+	join_candidates {
+		let x in 3..1_000;
+		// Worst Case Complexity is insertion into an ordered list so \exists full list before call
+		let mut candidate_count = 1u32;
+		for i in 2..x {
+			let seed = USER_SEED - i;
+			let collator = create_funded_collator::<T>(
+				"collator",
+				seed,
+				0u32.into(),
+				candidate_count
+			)?;
+			candidate_count += 1u32;
+		}
+		let caller: T::AccountId = create_funded_user::<T>("caller", USER_SEED, 0u32.into());
+	}: _(RawOrigin::Signed(caller.clone()), default_balance::<T>(), candidate_count)
+	verify {
+		assert!(Pallet::<T>::is_candidate(&caller));
+	}
+
+	// This call schedules the collator's exit and removes them from the candidate pool
+	// -> it retains the self-bond and nominator bonds
+	leave_candidates {
+		let x in 3..1_000;
```

Comment:
```suggestion
		let x in 3..1_000;
```

This implies `1_000` is the max collator candidates, but it isn't enforced. When I change `1_000` -> `u32::MAX` the benchmark just keeps running for a really long time (probably a few hours).

Reply:
Are you planning to address this in this PR? Seems pretty important :)

Reply:
Not sure how to address it, it's more of an observation.

It doesn't mean `1_000` is the max collator candidates allowed in the benchmark. It just means that 1_000 is the largest number of collators registered in the benchmark. I sort of assume that the benchmark generates a function that expresses a linear relationship with number of collator candidates and uses the number of collator candidates as an input to calculate the weight. The generated weight function is not bounded to 1_000 collator candidates so I expect it wouldn't change much by increasing the max to a higher number, but I don't really know. It is a good experiment to run.